From 935ded6b57d0bf6466e04946982b82c2dd3560b6 Mon Sep 17 00:00:00 2001
From: Jeremy Fitzhardinge <jeremy.fitzhardinge@citrix.com>
Date: Wed, 3 Nov 2010 16:40:48 -0400
Subject: [PATCH 19/26] xen/events: only unmask irq if enabled

A dynirq EOI is an unmask, but we should only unmask if the
irq is logically enabled.

Signed-off-by: Jeremy Fitzhardinge <jeremy.fitzhardinge@citrix.com>
(cherry picked from commit 2b1c9503c615f68262ae2e96ee26ee128b486287)

Signed-off-by: Ian Campbell <ijc@hellion.org.uk>
---
 drivers/xen/events.c |    3 ++-
 1 files changed, 2 insertions(+), 1 deletions(-)

diff --git a/drivers/xen/events.c b/drivers/xen/events.c
index e547c90..9ead93e 100644
--- a/drivers/xen/events.c
+++ b/drivers/xen/events.c
@@ -1199,10 +1199,11 @@ int resend_irq_on_evtchn(unsigned int irq)
 static void ack_dynirq(unsigned int irq)
 {
 	int evtchn = evtchn_from_irq(irq);
+	struct irq_desc *desc = irq_to_desc(irq);
 
 	move_masked_irq(irq);
 
-	if (VALID_EVTCHN(evtchn))
+	if (VALID_EVTCHN(evtchn) && !(desc->status & IRQ_DISABLED))
 		unmask_evtchn(evtchn);
 }
 
-- 
1.7.5.4

